### PR TITLE
mgr/dashboard: integrate progress mgr module events into dashboard tasks list

### PR DESCRIPTION
**Pull request by rjfd**
This PR integrates the executing and finished events from the progress mgr module into the list of dashboard tasks.
The progress module events are retrieved by the dashboard using the inter-module calls provided by ceph-mgr.

This allows to show in the dashboard frontend the long-running asynchronous actions registered in the progress module alongside with the dashboard tasks.

We also add to extend the information stored for the completed events in the progress module to allow them to be more "compatible" with the dashboard events.
See commits 7904b9e and 87469c0

Fixes: https://tracker.ceph.com/issues/38202

Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

**Comment by rjfd** [comment]
@liewegas can you take a look at the commits 7904b9e and 87469c0 , that make a small change to the progress module?

**Comment by liewegas** [comment]
@kamoltat this look ok to you too?

**Comment by liewegas** [comment]
> @liewegas can you take a look at the commits [7904b9e](https://github.com/ceph/ceph/commit/7904b9eb4ae39051d2ae68a86361be41052c679d) and [87469c0](https://github.com/ceph/ceph/commit/87469c0e6ed293a9bd3833cb2f881150eadc84a0) , that make a small change to the progress module?

LGTM!

**Comment by kamoltat** [comment]
LGTM too

> @kamoltat this look ok to you too?

LGTM!

**Comment by rjfd** [comment]
We can ignore the failures in QA tests. The dashboard tests have succeeded and the jobs are being marked as failed due to warning caused by recent changes to the volumes module.

Therefore this PR is ready to be merged.

File: src/pybind/mgr/dashboard/services/progress.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            'begin_time': "{}Z".format(datetime.fromtimestamp(ev["started_at"])
+                                       .isoformat()),
+            'progress': int(100 * ev['progress'])
+        })
+
+    for ev in progress_events['completed']:
+        logger.debug("[Progress] finished event=%s", ev)
+        finished_t.append({
+            'name': "progress/{}".format(ev['message']),
+            'metadata': dict(ev['refs']),
```
Comment (dillaman) [inline src/pybind/mgr/dashboard/services/progress.py]: You cannot be assured that the `refs` is in a `dict` format

Comment (rjfd) [reply src/pybind/mgr/dashboard/services/progress.py]: Indeed. But can we change that? We could agree on a list of pairs format, or even a dict.

Example:
`refs = [("origin", "rbd_support"), ...]`

Comment (dillaman) [reply src/pybind/mgr/dashboard/services/progress.py]: I'm fine w/ changing it. I had originally expected `refs` to be a free-form `dict` but then saw it was never used that way.

**Pull request merged**